application/include/mmul.hpp

```diff
@@ -17,16 +17,17 @@
 using namespace Eigen;
 
 template <typename Executor,
-          typename execution::instance_of<inline_executor, Executor> = 0>
+          typename execution::instance_of_base<inline_executor, Executor> = 0>
 void mmul(Executor ex, MatrixXd &a, MatrixXd &b, MatrixXd &c) {
-  c = a * b;
+  auto mul = [](MatrixXd &a, MatrixXd &b, MatrixXd &c) { c = a * b; };
+
+  ex.execute(std::bind(mul, std::ref(a), std::ref(b), std::ref(c)));
```

Comment:
`bind` ... over lambda? I do hope you plan on rectifying this 😄

Reply:
Fixed. Hadn't notice this.